src/entities/repos/submission_repo.py

```diff
@@ -14,46 +17,76 @@
     FilingDTO,
     FilingDAO,
     FilingTaskDAO,
+    FilingTaskStateDAO,
+    FilingTaskState,
 )
 
 logger = logging.getLogger(__name__)
 
 T = TypeVar("T")
 
 
+class NoFilingPeriodException(Exception):
+    pass
+
+
 async def get_submissions(session: AsyncSession, filing_id: int = None) -> List[SubmissionDAO]:
-    async with session.begin():
-        stmt = select(SubmissionDAO)
-        if filing_id:
-            stmt = stmt.filter(SubmissionDAO.filing == filing_id)
-        results = await session.scalars(stmt)
-        return results.all()
+    return await query_helper(session, SubmissionDAO, "filing", filing_id)
 
 
 async def get_filing_periods(session: AsyncSession) -> List[FilingPeriodDAO]:
-    async with session.begin():
-        stmt = select(FilingPeriodDAO)
-        results = await session.scalars(stmt)
-        return results.all()
+    return await query_helper(session, FilingPeriodDAO)
 
 
 async def get_submission(session: AsyncSession, submission_id: int) -> SubmissionDAO:
-    return await query_helper(session, submission_id, SubmissionDAO)
+    result = await query_helper(session, SubmissionDAO, "id", submission_id)
+    return result[0] if result else None
 
 
 async def get_filing(session: AsyncSession, filing_id: int) -> FilingDAO:
-    return await query_helper(session, filing_id, FilingDAO)
+    result = await query_helper(session, FilingDAO, "id", filing_id)
+    result = deepcopy(result)
+    if result:
+        await populate_missing_tasks(session, result)
+    return result[0] if result else None
+
+
+async def get_period_filings_for_user(
+    session: AsyncSession, user: AuthenticatedUser, period_name: str
+) -> List[FilingDAO]:
+    filing_period = await query_helper(session, FilingPeriodDAO, "name", period_name)
+    if filing_period:
+        filings = await query_helper(session, FilingDAO, "filing_period", filing_period[0].id)
+        filings = [f for f in filings if f.lei in user.institutions]
+        filing_leis = [f.lei for f in filings]
+        missing_filing_leis = [lei for lei in user.institutions if lei not in filing_leis]
+        if missing_filing_leis:
+            for lei in missing_filing_leis:
+                filing_dao = await upsert_filing(
+                    session,
+                    FilingDTO(
+                        lei=lei,
+                        tasks=[],
+                        filing_period=filing_period[0].id,
+                        institution_snapshot_id="v1",  # TODO: add function to get this from user-fi-api
+                    ),
+                )
+                filings.append(filing_dao)
```

Comment:
not sure how I feel about creating filings just by the act of user retrieving filings for the period; let's bring it during the dev touch point.

Reply:
Ok removed the creating of the 'missing' FilingDAO.  Wrote https://github.com/cfpb/sbl-filing-api/issues/64 to refactor based on the discussion of lei+period_name PKs.